dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,385 @@
+import math
+import numpy as np
+import scipy.stats as stats
+from functools import partial
+from pprint import pprint
+from toolz import reduce
+import toolz
+import warnings
+from timeit import default_timer
+
+import sklearn
+from dask_ml.model_selection._split import train_test_split
+
+from sklearn import clone
+from sklearn.model_selection import ParameterSampler
+
+import dask.array as da
+from .model_selection import DaskBaseSearchCV, GridSearchCV, _RETURN_TRAIN_SCORE_DEFAULT
+
+
+def _train(model, X, y, X_val, y_val, max_iter=1, dry_run=False, classes=None,
+           s=1, i=1, k='_', verbose=False, **fit_kwargs):
+    """
+    Train function. Returns validation loss.
+
+    Takes max_iters as parameters. THese are treated as a "scarce resource" --
+    Hyperband is trying to minimize the total iters use for max_iters.
+    """
+    start_time = default_timer()
+    for iter in range(int(max_iter)):
+        if verbose:
+            msg = ("Training model {k} in bracket s={s} iteration {i}. "
+                   "This model is {percent:.1f}% trained for this bracket")
+            print(msg.format(iter=iter, k=k, s=s, i=i, percent=iter * 100.0 / max_iter))
```

Comment:
Let's try to use the `logging` module, as it's more useful in a distributed setting (`print` statements may be hard / difficult to access if they're on a different machine).

At the top of the module, you can use `logger = logging.getLogger(__name__)` to get or create a new logger with the name `dask_search_cv.adaptive`. Then when you actually want to see the log, you can do an annoying

```python
logger = logging.getLogger('dask_searchcv.adaptive')
logger.setLevel(logging.INFO)
handler = logging.StreamHandler()
handler.setLevel(logging.INFO)
logger.addHandler(handler)
```

There's probably a better way to integrate with `distributed` here.

Reply:
I'm running into an issue, it's not logging on this line in particular (though it's logging on other lines). I'll try to formulate a bug report.